src/routes.py

```diff
@@ -122,12 +122,12 @@ def topic_page(topic_id: int) -> str | Response:
         user_session = UserSession.get_user_session_by_session_id(session_id)
 
     current_user = None
-    if user_session is not None:
-        current_user = user_session.user
-        if not (topic := Topic.get(topic_id)):
-            return render_template("404.html", current_user=current_user)
+    if user_session is None:
+        return redirect(url_for("routes.login"))
+    current_user = user_session.user
+    if (topic := Topic.get(topic_id)):
```

Comment:
Let's remain this line as it was. Looks like this change isn't related to the ticket (except the indentation of course).

And we also have such line in `create_post`:
```python
if not (topic := Topic.get(topic_id)):
```
so if we will remain line as it was, it will be more consistent